utils/mysql/files/mysqld.init

```diff
@@ -7,24 +7,31 @@ STOP=10
 SERVICE_DAEMONIZE=1
 SERVICE_WRITE_PID=1
 SERVICE_STOP_TIME=9
-
-error() {
-	echo "${initscript}:" "$@" 1>&2
-}
+PROG='/usr/bin/mysqld'
 
 start() {
-	local datadir=$(sed -n -e "s/^[[:space:]]*datadir[[:space:]]*=[[:space:]\"']*\([^[:space:]\"']*\)[[:space:]\"']*/\1/p" /etc/my.cnf)
```

Comment:
An alternative suggestion that works for all cases:
`sed -n -e "s/^[ \t]*datadir[ \t]*=[ \t]*\(\(.*[^ \t]\)\|'\([^']*\)'\|\"\([^\"]*\)\"\)[ \t]*\$/\2\3\4/p"`

This properly extracts paths for all these cases:

    datadir = foo
    datadir = "foo bar"
    datadir = 'bar "lala" baz'
    datadir = path with spaces